front/components/providers/AzureOpenAISetup.jsx

```diff
@@ -0,0 +1,194 @@
+import { Fragment, useState } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+import { ActionButton, Button } from "@app/components/Button";
+import { useSWRConfig } from "swr";
+import { checkProvider } from "@app/lib/providers";
+
+export default function AzureOpenAISetup({ open, setOpen, config, enabled }) {
+  const { mutate } = useSWRConfig();
+
+  const [apiKey, setApiKey] = useState(config ? config.api_key : "");
+  const [endpoint, setEndpoint] = useState(config ? config.endpoint : "");
+  const [testSuccessful, setTestSuccessful] = useState(false);
+  const [testRunning, setTestRunning] = useState(false);
+  const [testError, setTestError] = useState("");
+  const [enableRunning, setEnableRunning] = useState(false);
+
+  if (config && config.api_key.length > 0 && apiKey.length == 0) {
+    setApiKey(config.api_key);
```

Comment:
Calling a `setState()` function inside a render function is not allowed by React.
It can work but it can also lead to infinite render loops or a lot unnecessary render calls.


The react way to do that is :
```
  useEffect(() => {
    if (apiKey.length == 0 && config.api_key.length > 0) {
      setApiKey(config.api_key);
    }
  }, [apiKey, config.api_key]);
```

`useEffect()` is allowed to perform `setState()` calls  and the dependency array at the end make it render just the right number of time.

Reply:
Awesome. Will address :+1: